Sources/SwiftGenKit/Parsers/Strings/FileTypeParser/StringsFileParser.swift

```diff
@@ -23,100 +23,96 @@ extension Strings {
         throw ParserError.failureOnLoading(path: path)
       }
 
-      let entries = try PropertyListDecoder()
+      var entries = try PropertyListDecoder()
         .decode([String: String].self, from: data)
-        .map { key, translation -> (String, Entry) in
-          let entry = try Entry(key: key, translation: translation, keyStructureSeparator: options[Option.separator])
-          return (key, entry)
+        .map { key, translation -> Entry in
+          try Entry(key: key, translation: translation, keyStructureSeparator: options[Option.separator])
         }
 
-      var dict = Dictionary(uniqueKeysWithValues: entries)
+      parseComments(path: path, entries: &entries)
 
-      if let string: String = try? path.read() {
-        let scanner = Scanner(string: string)
-        while !scanner.isAtEnd {
-          let comment = scanner.scanComment()
-          if let key = scanner.scanKey() {
-            dict[key]?.comment = comment
-          }
-        }
+      return entries
+    }
+
+    /// Try to parse the comments in a strings file, and add them to the corresponding entries.
+    ///
+    /// First we remove all keys & values, and replace them by "magic markers", so that we can more easily parse the
+    /// comments. Then we try to parse a comment, and then a marker, and repeat this until we're done.
+    func parseComments(path: Path, entries: inout [Entry]) {
+      guard var contents: String = try? path.read() else { return }
+
+      // replace entries with markers
+      for (index, entry) in zip(0..., entries) {
+        let key = entry.key.escapedStringsRegex
+        let translation = entry.translation.escapedStringsRegex
+        contents = contents.replacingOccurrences(
+          of: #"(?m)"\#(key)"\s*=\s*"\#(translation)"\s*;"#,
+          with: "\(String.startMarker)\(index)\(String.endMarker)",
+          options: .regularExpression
+        )
```

Comment:
The whole logic of the "marker system" will also fail for any `/* "key" = "translation" */` occurrence. For example:

```
"key1" = "value1";

// Same as "key1" = "value1"; but in the context of user not logged in
"key1.anonymous" = "value2";
```

imho, it's better to implement a proper tokenizer rather than trying to play with this.